src/extension/background.js

```diff
@@ -627,13 +627,30 @@ const internalActions = {
   },
 };
 
+function sendMessageToTabs(tabs) {
+  for (const tab of tabs) {
+    chrome.tabs
+      .sendMessage(tab.id, { action: 'closePalette' });
+  }
+}
+
 /**
  * Adds the listeners for the extension.
  */
 (async () => {
   // external messaging API for projects to communicate with sidekick
   chrome.runtime.onMessageExternal.addListener(async (message, sender, sendResponse) => {
     const { action } = message;
+    if (action === 'closePalette') {
```

Comment:
Looking at the code, I now realize we could actually move this down into the `externalActions` object instead, and follow that pattern.

Defining:
```js
// loads the sidekick if the project is configured
  closePalette: async (_, { tab }) => {
    …
    return Promise.resolve();
  },
```

And since we now have the source tab, you also don't need to iterate over all tabs and could get rid of `sendMessageToTabs` method, and directly just call `.sendMessage`

Reply:
I agree, let's move this logic into `externalActions`. And wrap `sendMessageToTabs` into that handler as well as long as it's not used anywhere else yet.